WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -953,14 +957,21 @@ private async Task<IEnumerable<TxOut>> ProceedWithOutputRegistrationPhaseAsync(u
 		// Calculate outputs values
 		var constructionState = roundState.Assert<ConstructionState>();
 
-		AmountDecomposer amountDecomposer = new(roundParameters.MiningFeeRate, roundParameters.AllowedOutputAmounts, Constants.P2wpkhOutputVirtualSize, Constants.P2wpkhInputVirtualSize, (int)availableVsize);
+		// Get the output's size and its of the input that will spend it in the future.
+		// Here we assume all the outputs share the same scriptpubkey type.  
+		var preferTaprootOutputs = false;
+		var (inputVirtualSize, outputVirtualSize)= DestinationProvider.Peek(preferTaprootOutputs).IsScriptType(ScriptType.Taproot)
```

Comment:
Good question. First of all, and before answering your question, let agree that there is a `preferTaprootOutputs = false`  that forces the selection of segwit always so, about your latter question:

> "Why aren't we using SegWit as output here?" 

We are using segwit always.

> Why is it that we are basing the decision on "the last generated address"?

It is a super-convoluted way to use segwit only. It is there because I needed to test taproot scripts in coinjoins and implement the changes to make coinjoins work. I disabled that and then made the script type selection more restrictive.

I can hardcode the following and it will work exactly the same way:

	var (inputVirtualSize, outputVirtualSize) = (Constants.P2wpkhInputVirtualSize, Constants.P2wpkhOutputVirtualSize);

And I can add this code back after fixing the `AmountDecomposer`.

## Context

In order to support taproot in a completely transparent way for our users nopara73 suggested to use those script types for **internal** keys only. Internal keys are used for two things: change outputs and coinjoin outputs. However, that's not the only option; we could use internal taproot for coinjoin outputs only and use internal segwit for change outputs only what could make the code a bit easier/cleaner too. 

The  thing is that because of the `AmountDecomposer`  cannot work with different script types yet, all the output scripts must share the same script type (all segwit or all taproot). And that means that we need to know script type in advance. The previous version was consuming all  the available segwit keys before starting to use taproot scripts, while this version uses only segwit no matter what.

Reply:
ACK, added notes to here: https://github.com/zkSNACKs/WalletWasabi/issues/9622